rest_framework_json_api/relations.py

```diff
@@ -27,6 +27,7 @@
     "related_link_view_name",
     "related_link_lookup_field",
     "related_link_url_kwarg",
+    "related_link_related_name",
```

Comment:
> This fixes the fact that getting related view's serializer gets it from the super class, rather than the current class.

The PR description states that this PR is addressing issue that related view does not properly get parent serializer class (it seems to be another clean up after #860). I guess [this change](https://github.com/django-json-api/django-rest-framework-json-api/pull/904/files#diff-f6ec81f584f0c3b87cc8ba41a977e3631365ab6242c69f1ec9db51916d65a588) addresses this issue.

So at a quick glance for me it is not quite clear why introducing of a new links parameters is needed to fix this issue. Can you elaborate?

Reply:
Explanation 2:
We need to add this parameter to specify the the `related` link for hyperlink fields (which can, most of the time, have a different related name).
For example in the example serializers, the `field_name` is `comments_hyperlinked`:
```python
comments_hyperlinked = relations.HyperlinkedRelatedField(
        related_link_view_name="entry-comments",
```
And the related view is:
```python
    url(
        r"entries/(?P<entry_pk>[^/.]+)/comments$",
        CommentViewSet.as_view({"get": "list"}),
        name="entry-comments",
    ),
```
So `comments_hyperlinked` relates to `comments`.
But [here](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/rest_framework_json_api/relations.py#L116), we get the field_names, which will create such a url: `/entries/1/comments_hyperlinked`, which isn't really what we want.